aws/resource_aws_storagegateway_fsx_associate_file_system_test.go

```diff
@@ -0,0 +1,436 @@
+					resource.TestCheckResourceAttrPair(resourceName, "audit_destination_arn", "aws_cloudwatch_log_group.test2", "arn"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSStorageGatewayFileSystemAssociation_disappears(t *testing.T) {
+	var fsxFileSystemAssociation storagegateway.FileSystemAssociationInfo
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_storagegateway_file_system_association.test"
+	domainName := testAccRandomDomainName()
+	username := "Admin"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck(storagegateway.EndpointsID, t) },
+		ErrorCheck:   testAccErrorCheck(t, storagegateway.EndpointsID),
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsStorageGatewayFsxAssociateFileSystemDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsStorageGatewayFileSystemAssociationConfig_Required(rName, domainName, username),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsStorageGatewayFsxAssociateFileSystemExists(resourceName, &fsxFileSystemAssociation),
+					testAccCheckResourceDisappears(testAccProvider, resourceAwsStorageGatewayFileSystemAssociation(), resourceName),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSStorageGatewayFileSystemAssociation_disappears_storageGateway(t *testing.T) {
+	var fsxFileSystemAssociation storagegateway.FileSystemAssociationInfo
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_storagegateway_file_system_association.test"
+	domainName := testAccRandomDomainName()
+	username := "Admin"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck(storagegateway.EndpointsID, t) },
+		ErrorCheck:   testAccErrorCheck(t, storagegateway.EndpointsID),
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsStorageGatewayFsxAssociateFileSystemDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsStorageGatewayFileSystemAssociationConfig_Required(rName, domainName, username),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsStorageGatewayFsxAssociateFileSystemExists(resourceName, &fsxFileSystemAssociation),
+					testAccCheckResourceDisappears(testAccProvider, resourceAwsStorageGatewayGateway(), "aws_storagegateway_gateway.test"),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSStorageGatewayFileSystemAssociation_disappears_fsxFileSystem(t *testing.T) {
+	var fsxFileSystemAssociation storagegateway.FileSystemAssociationInfo
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_storagegateway_file_system_association.test"
+	domainName := testAccRandomDomainName()
+	username := "Admin"
+
+	t.Skip("A bug in the service API has been reported. Deleting the FSx file system before the association prevents association from being deleted.")
```

Comment:
The linter complains here because the variables are declared before the `t.Skip()`, but only used after it. If you move the `t.Skip()` above the declarations, it will pass.